dashboard/src/services/billing.ts

```diff
@@ -0,0 +1,65 @@
+import { getUserSubscription } from '@/repositories/postgres/subscription';
+import { getUserSiteIds } from '@/repositories/postgres/dashboard';
+import { getUserEventCountForPeriod } from '@/repositories/clickhouse/usage';
+import { getBillingHistoryByUserId } from '@/repositories/postgres/billingHistory';
+import { toDateString } from '@/utils/dateFormatters';
+import {
+  UserBillingDataSchema,
+  type UsageData,
+  type UserBillingData,
+  type BillingHistory,
+} from '@/entities/billing';
+
+export async function getUserBillingStats(userId: string): Promise<UserBillingData> {
+  try {
+    const subscription = await getUserSubscription(userId);
+    if (!subscription) {
+      throw new Error('No subscription found for user');
+    }
+
+    const siteIds = await getUserSiteIds(userId);
+
+    const currentUsage = await getUserEventCountForPeriod(siteIds, toDateString(subscription.currentPeriodStart));
+
+    const usage: UsageData = {
+      current: currentUsage,
+      limit: subscription.eventLimit,
+      remaining: Math.max(0, subscription.eventLimit - currentUsage),
+      isOverLimit: currentUsage >= subscription.eventLimit,
```

Comment:
Nitpick: A user is technically first over limit once they've passed the event limit
```suggestion
      isOverLimit: currentUsage > subscription.eventLimit,
```